inverse_covariance/two_way_standard_scaler.py

```diff
@@ -0,0 +1,329 @@
+        The mean value for each feature in the training set.
+
+    row_var_ : array of floats with shape [n_examples]
+        The variance for each feature in the training set. Used to compute
+        `scale_`
+
+    col_var_ : array of floats with shape [n_features]
+        The variance for each feature in the training set. Used to compute
+        `scale_`
+
+    n_samples_seen_ : int
+        The number of samples processed by the estimator. Will be reset on
+        new calls to fit, but increments across ``fit`` calls.
+
+    See also
+    --------
+    :class:`sklearn.preprocessing.StandardScaler`
+    :class:`sklearn.decomposition.PCA`
+
+    Notes
+    -----
+    See the implications of one-way vs. two-way standardization in here. TBD
+    """
+
+    def __init__(self, copy=True, with_mean=True, with_std=True):
+        """Unlike StandardScaler, with_mean is always set to True, to ensure
+        that two-way standardization is always performed with centering. The
+        argument `with_mean` is retained for the sake of sklearn
+        API compatibility.
+        """
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.copy = copy
+
+    def _reset(self):
+        fit_attrs = [
+            "n_rows_",
+            "n_cols_",
+            "col_var_",
+            "col_mean_",
+            "row_var_",
+            "row_mean_",
+            "row_scale_",
+            "col_scale_",
+            "n_rows_seen_",
+            "n_cols_seen_",
+        ]
+        for attr in fit_attrs:
+            if hasattr(self, attr):
+                delattr(self, attr)
+
+    def _initial_state(self):
+        return not hasattr(self, "n_rows_seen_")
+
+    def fit(self, X, y=None):
+        """Compute the mean and std for both row and column dimensions.
+
+        Parameters
+        ----------
+        X : {array-like}, shape [n_rows, n_cols]
+            The data used to compute the mean and standard deviation
+            along both row and column axes
+
+        y : Passthrough for ``Pipeline`` compatibility. Input is ignored.
+        """
+        # Reset internal state before fitting
+        self._reset()
+        return self.partial_fit(X, y)
+
+    def partial_fit(self, X, y=None):
+        """Online computation of mean and std on X for later scaling."""
+        X = check_array(
+            X,
+            accept_sparse=False,
+            copy=self.copy,
+            warn_on_dtype=True,
+            dtype=FLOAT_DTYPES,
+            estimator=self,
+            ensure_min_features=2,
+        )
+        if sparse.issparse(X):
+            raise NotImplemented(
+                "Input is sparse: Algorithm for sparse matrices currently not supported."
+            )
+
+        self.n_rows_, self.n_cols_ = np.shape(X)
+
+        # Q: This doesnt seem to actually get used in the transform, only in
+        #    the inverse transform which it sounds like we should not support.
+
```

Comment:
@mnarayan The values computed in the fit/partial_fit routines do not get used in the two_way_standard_scaler routine, only in the "inverse transform" method.  Do we actually need them or are they just cribbed from sklearn?

Reply:
@mnarayan did you see this question about partial fit?  If these variables are not needed, we also don't need partial_fit, can simplify everything, and have this be a transform-only transformer.